tidy3d/components/medium.py

```diff
@@ -5180,6 +5181,190 @@ class SurfaceImpedanceFitterParam(Tidy3dBaseModel):
     )
 
 
+class AbstractSurfaceRoughness(Tidy3dBaseModel):
+    """Abstract class for modelling surface roughness of lossy metal."""
+
+    @abstractmethod
+    def roughness_correction_factor(self, frequency: float, skin_depths: float) -> complex:
+        """Complex-valued roughness correction factor applied to surface impedance.
+
+        Notes
+        -----
+            The roughness correction factor should be causal. It is multiplied to the
+            surface impedance of the lossy metal to account for the effects of surface roughness.
+
+        Parameters
+        ----------
+        frequency : float
+            Frequency to evaluate roughness correction factor at (Hz).
+        skin_depths : float
+            Skin depths of the lossy metal that is frequency-dependent.
```

Comment:
these can be arrays, right?

Reply:
yes

Reply:
But I'm following the convention in this file, e.g. `eps_model(self, frequency: float)` can take arrays too.

Reply:
I think it was unintentional there https://github.com/flexcompute/tidy3d/issues/1372

Reply:
ha, i see. Forgot that it was assigned to me!